tests/fields.go

```diff
@@ -85,16 +85,24 @@ func (ps *ProcessorSetup) EventFieldsMappedToTemplateFields(t *testing.T, eventF
 	for _, val := range eventFields.Array() {
 		var f = val.(string)
 		for _, m := range mappings {
-			if strings.HasPrefix(f, m.Template) {
-				f = strings.Replace(f, m.Template, m.Mapping, -1)
+			template := m.Template
```

Comment:
It took me awhile to come around but I like this, certainly better than the alternative of not checking that the field is present at all.

Reply:
it took me quite a while to figure out why the test was even failing... i also think is better checking than not, but implementation is awfully `if`-fy. If you have any better ideas, happy to hear...

Reply:
This is the alternative I had considered, removing labels.* from the check altogether.  I like your solution better.  We have discussed overhauling these tests because of this kind of thing, I'm happy to leave this as is and tackle it later during that effort.

```patch
diff --git a/processor/stream/package_tests/metadata_attrs_test.go b/processor/stream/package_tests/metadata_attrs_test.go
index 0c45338b..2876a9aa 100644
--- a/processor/stream/package_tests/metadata_attrs_test.go
+++ b/processor/stream/package_tests/metadata_attrs_test.go
@@ -92,7 +92,7 @@ func getMetadataEventAttrs(t *testing.T, prefix string) *tests.Set {
        contextMetadata := metadata["metadata"]
 
        eventFields := tests.NewSet()
-       tests.FlattenMapStr(contextMetadata, prefix, nil, eventFields)
+       tests.FlattenMapStr(contextMetadata, prefix, tests.NewSet(tests.Group("labels")), eventFields)
        t.Logf("Event fields: %s", eventFields)
        return eventFields
 }
@@ -100,7 +100,6 @@ func getMetadataEventAttrs(t *testing.T, prefix string) *tests.Set {
 func TestMetadataPayloadAttrsMatchFields(t *testing.T) {
        setup := metadataProcSetup()
        eventFields := getMetadataEventAttrs(t, "")
-
        var mappingFields = []tests.FieldTemplateMapping{
                {Template: "system.container.", Mapping: "container."},             // move system.container.*
                {Template: "system.container", Mapping: ""},                        // delete system.container
@@ -115,7 +114,6 @@ func TestMetadataPayloadAttrsMatchFields(t *testing.T) {
                {Template: "service.agent", Mapping: "agent"},
                {Template: "user.username", Mapping: "user.name"},
                {Template: "process.argv", Mapping: "process.args"},
-               {Template: "labels.*", Mapping: "labels"},
        }
        setup.EventFieldsMappedToTemplateFields(t, eventFields, mappingFields)
 }
@@ -123,8 +121,8 @@ func TestMetadataPayloadAttrsMatchFields(t *testing.T) {
 func TestMetadataPayloadMatchJsonSchema(t *testing.T) {
        metadataProcSetup().AttrsMatchJsonSchema(t,
                getMetadataEventAttrs(t, ""),
-               tests.NewSet(tests.Group("labels")),
                nil,
+               tests.NewSet(tests.Group("labels")),
        )
 }
```